src/Pages/listClient.tsx

```diff
@@ -1,111 +1,124 @@
-import ButtonPrimary from "@/components/buttonPrimary/buttonPrimary";
 import Card from "@/components/Cards/CardUser";
+import DrawerClient from "@/components/drawerClient/drawerClient";
 import Spinner from "@/components/spinner/spinner";
-import useAuthentication from "@/hook/useAuthentication";
 import { SearchField } from "@/hook/useClient/types";
 import useClient from "@/hook/useClient/useClient";
 import { useClientSearch } from "@/hook/useClient/useSearchClient";
+import useDebounce from "@/hook/useDebounce/useDebounce";
+import useResizeObserver from "@/hook/useResizeObserver";
+import useRowVirtualizer from "@/hook/useRowVirtualizer";
 import { Client } from "@/types";
 import { useState } from "react";
-import { Link } from "react-router-dom";
+import SearchInput from "@/components/searchInput/searchInput";
 
 const ListClient = () => {
-  useAuthentication();
-
   const [searchType, setSearchType] = useState<SearchField>("name");
   const [searchValue, setSearchValue] = useState("");
-  const [searchActive, setSearchActive] = useState(false);
-
-  const { clients, isLoading } = useClient();
+  const { clients, isLoading, fetchNextPage, hasNextPage, isFetchingNextPage } =
+    useClient();
+  const [debouncedSearch] = useDebounce(searchValue, 400);
 
-  const searchParams = { [searchType]: searchValue.trim() };
-  const isSearchEnabled = searchActive && !!searchValue.trim();
+  const searchParams = { [searchType]: debouncedSearch };
 
-  const { data: searchResults, isLoading: isSearching } = useClientSearch({
+  const { data: SearchClients } = useClientSearch({
     searchTerm: searchParams,
-    enabled: isSearchEnabled,
+    enabled: !!debouncedSearch.trim(),
```

Comment:
É uma boa prática não utilizar negação dupla, geralmente deixa o cógigo mais difícil de interpretar.
```suggestion
    enabled: Boolean(debouncedSearch.trim()),
```